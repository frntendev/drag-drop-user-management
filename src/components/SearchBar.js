import React, { Component } from "react";
import { FormControl, FormGroup } from "material-ui/Form";
import TextField from "material-ui/TextField";
import { css } from "react-emotion";
class SearchBar extends Component {
  constructor(props) {
    super(props);
    this.state = { searchTerm: "" };
  }

  onInputChange(event) {
    this.setState({ searchTerm: event.target.value }, () => {
      this.props.onSearchSubmit(this.state.searchTerm);
    });
  }

  onFormSubmit(event) {
    event.preventDefault();
    this.props.onSearchSubmit(this.state.searchTerm);
  }

  onFormReset() {
    this.props.onSearchReset();
    this.setState({ searchTerm: "" });
  }

  render() {
    return (
      <FormControl
        id="search-form"
        component="form"
        onSubmit={this.onFormSubmit.bind(this)}
        onReset={this.onFormReset.bind(this)}
        className={css`
          width: 100%;
          @media (max-width: 800px) {
            padding-left: 24px;
            padding-right: 24px;
            box-sizing: border-box;
          }
        `}
      >
        <FormGroup>
          <TextField
            id="search-input"
            label="Search"
            value={this.state.searchTerm}
            margin="normal"
            onChange={this.onInputChange.bind(this)}
            fullWidth
          />
        </FormGroup>
      </FormControl>
    );
  }
}

export default SearchBar;
