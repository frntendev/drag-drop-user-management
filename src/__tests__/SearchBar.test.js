import React from "react";
import { shallow, mount, configure } from "enzyme";
import { Field } from "redux-form";
import SearchBar from "../components/SearchBar";
import Adapter from "enzyme-adapter-react-15";
import renderer from "react-test-renderer";

configure({ adapter: new Adapter() });

describe("SearchBar Component", () => {
  const shallowComponent = shallow(<SearchBar />);
  const mountComponent = mount(<SearchBar />);
  it("should render without throwing an error", () => {
    expect(shallowComponent).toMatchSnapshot();
  });
  it("should find textfields in the component", () => {
    expect(mountComponent.find("#search-input").length).toEqual(4);
  });
  it("should respond to change event and change the state of the SearchBar Component", () => {
    const wrapper = shallow(
      <SearchBar onSearchSubmit={() => console.log("Yeap!")} />
    );
    wrapper
      .find("#search-input")
      .simulate("change", { target: { value: "sep" } });
    expect(wrapper.state("searchTerm")).toEqual("sep");
  });
});
