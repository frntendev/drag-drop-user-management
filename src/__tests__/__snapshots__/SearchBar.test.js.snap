// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar Component should render without throwing an error 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchBar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <WithStyles(FormGroup)>
        <TextField
          fullWidth={true}
          id="search-input"
          label="Search"
          margin="normal"
          onChange={[Function]}
          required={false}
          select={false}
          value=""
        />
      </WithStyles(FormGroup)>,
      "className": "css-dk14do",
      "component": "form",
      "id": "search-form",
      "onReset": [Function],
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <TextField
          fullWidth={true}
          id="search-input"
          label="Search"
          margin="normal"
          onChange={[Function]}
          required={false}
          select={false}
          value=""
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "fullWidth": true,
          "id": "search-input",
          "label": "Search",
          "margin": "normal",
          "onChange": [Function],
          "required": false,
          "select": false,
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <WithStyles(FormGroup)>
          <TextField
            fullWidth={true}
            id="search-input"
            label="Search"
            margin="normal"
            onChange={[Function]}
            required={false}
            select={false}
            value=""
          />
        </WithStyles(FormGroup)>,
        "className": "css-dk14do",
        "component": "form",
        "id": "search-form",
        "onReset": [Function],
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <TextField
            fullWidth={true}
            id="search-input"
            label="Search"
            margin="normal"
            onChange={[Function]}
            required={false}
            select={false}
            value=""
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "fullWidth": true,
            "id": "search-input",
            "label": "Search",
            "margin": "normal",
            "onChange": [Function],
            "required": false,
            "select": false,
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
